rhtap.Dockerfile

```diff
@@ -0,0 +1,135 @@
+ARG BUILD_DIR=/build
+ARG SRC_ROOT_DIR=${BUILD_DIR}
+ARG CMAKE_BUILD_DIR=${SRC_ROOT_DIR}/cmake-build
+ARG USE_VALGRIND=false
+ARG ADDRESS_SANITIZER=false
+
+# TODO: follow tags
+# TODO: use entitled RHEL/ubi to access cmake and other devel tools
+# FROM registry.access.redhat.com/ubi8/ubi:latest AS builder
+FROM quay.io/centos/centos:stream8 AS builder
+
+# TODO: reduce COPY scope
+COPY . .
+
+# TODO: prefetch dependencies for hermetic builds
+RUN dnf -y update && \
+    dnf -y install \
+        make \
+        wget \
+        unzip \
+        clang \
+        bpftool \
+        cmake-3.18.2-9.el8 \
+        gcc-c++ \
+        openssl-devel \
+        ncurses-devel \
+        curl-devel \
+        libuuid-devel \
+        libcap-ng-devel \
+        autoconf \
+        libtool \
+        git \
+        elfutils-libelf-devel \
+        tbb-devel \
+        # jq-devel # TODO: if entitlement/RHEL base image is solved, this can be installed from default repository \
+        c-ares-devel \
+    && \
+      # TODO: if entitlement/RHEL base image is solved, this can be installed from default repository \
+      dnf --enablerepo=powertools -y install jq-devel \
+    && \
+    dnf clean all
+
+ARG BUILD_DIR
+ENV BUILD_DIR=${BUILD_DIR}
+
+ARG SRC_ROOT_DIR
+ENV SRC_ROOT_DIR=${SRC_ROOT_DIR}
+
+ARG CMAKE_BUILD_DIR
+ENV CMAKE_BUILD_DIR=${CMAKE_BUILD_DIR}
+
+# TODO: CMAKE_BUILD_TYPE should probably not be Release for PR, normal branch builds
+ARG CMAKE_BUILD_TYPE=Release
+ENV CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+
+# TODO: what does this do?
```

Comment:
This is used on PRs, it appends an argument to the driver download URL that is used for filtering alerts on missing kernels. We haven't really been using it lately, but the TL;DR is that this needs to be true on PRs only.

Reply:
Added comment, resolution is captured under https://issues.redhat.com/browse/ROX-20240.